pkg/k8s/k8s_cluster.go

```diff
@@ -222,6 +225,26 @@ func (k *K8sCluster) GetObjectsByRefs(refs []k8s.ObjectRef) ([]*uo.UnstructuredO
 	return ret, retApiWarnings, nil
 }
 
+func (k *K8sCluster) GetObjectSource(ref k8s.ObjectRef) (string, string, []ApiWarning, error) {
```

Comment:
This function is very specific to flux, I would suggest to move it out of K8sCkuster and instead put it into cmd_flux.go as a simple helper function.